read-cache.c

```diff
@@ -611,6 +611,28 @@ int index_name_pos(struct index_state *istate, const char *name, int namelen)
 	return index_name_stage_pos(istate, name, namelen, 0);
 }
 
+int index_entry_exists(struct index_state *istate, const char *name, int namelen)
```

Comment:
This implementation _could_ be `return index_name_stage_pos() >= 0;` but we specifically don't want to expand the index in the case of a sparse directory. Perhaps it would be better to extend `index_name_stage_pos()` to have an extra parameter that controls that expansion. Then `index_entry_exists()` could still use that same binary search.

It would be worth commenting on the declaration of `index_entry_exists()` that it does not expand a sparse index and hence will return 0 for paths within sparse directory entries.

Reply:
Thanks, that certainly helped clean up some code duplication!